Pomodoro/Sources/App/ViewController.swift

```diff
@@ -6,15 +6,243 @@
+    private func drawBackgroundLayer() {
+        backgroundProgressLayer.path = UIBezierPath(
+            arcCenter: CGPoint(x: view.frame.midX, y: view.frame.midY),
+            radius: 150,
+            startAngle: -90.degreesToRadians,
+            endAngle: 270.degreesToRadians,
+            clockwise: true
+        ).cgPath
+        backgroundProgressLayer.strokeColor = UIColor.lightGray.cgColor
+        backgroundProgressLayer.fillColor = UIColor.clear.cgColor
+        backgroundProgressLayer.lineWidth = 10
+        view.layer.addSublayer(backgroundProgressLayer)
+    }
+
+    /// Foreground circle progress bar
+    private func drawForegroundLayer() {
+        let animationColor = isWorkTime ? UIColor.red : UIColor.green
+        foregroundProgressLayer.path = UIBezierPath(
+            arcCenter: CGPoint(x: view.frame.midX, y: view.frame.midY),
+            radius: 150,
+            startAngle: -90.degreesToRadians,
+            endAngle: 270.degreesToRadians,
+            clockwise: true
+        ).cgPath
+        foregroundProgressLayer.strokeColor = animationColor.cgColor
+        foregroundProgressLayer.fillColor = UIColor.clear.cgColor
+        foregroundProgressLayer.lineWidth = 8
+        view.layer.addSublayer(foregroundProgressLayer)
+    }
+
+    private func startResumeAnimation() {
+        !isAnimationStarted ? startAnimation() : resumeAnimation()
+    }
+
+    private func startAnimation() {
+        resetAnimation()
+        foregroundProgressLayer.strokeEnd = 0.0
+        animation.keyPath = "strokeEnd"
+        animation.fromValue = 0
+        animation.toValue = 1
+        animation.duration = Double(time)
+        animation.delegate = self
+        animation.isRemovedOnCompletion = false
+        animation.isAdditive = true
+        animation.fillMode = CAMediaTimingFillMode.forwards
+        foregroundProgressLayer.add(animation, forKey: "strokeEnd")
+        isAnimationStarted = true
+    }
+
+    private func resetAnimation() {
+        foregroundProgressLayer.speed = 1.0
+        foregroundProgressLayer.timeOffset = 0.0
+        foregroundProgressLayer.beginTime = 0.0
+        foregroundProgressLayer.strokeEnd = 0.0
+        isAnimationStarted = false
+    }
+
+    private func pauseAnimation() {
+        let pausedTime = foregroundProgressLayer.convertTime(CACurrentMediaTime(), from: nil)
+        foregroundProgressLayer.speed = 0.0
+        foregroundProgressLayer.timeOffset = pausedTime
+    }
+
+    private func resumeAnimation() {
+        let pausedTime = foregroundProgressLayer.timeOffset
+        foregroundProgressLayer.speed = 1.0
+        foregroundProgressLayer.timeOffset = 0.0
+        foregroundProgressLayer.beginTime = 0.0
+        let timeSincePaused = foregroundProgressLayer.convertTime(CACurrentMediaTime(), from: nil) - pausedTime
+        foregroundProgressLayer.beginTime = timeSincePaused
     }
 
+    private func stopAnimation() {
+        foregroundProgressLayer.speed = 1.0
```

Comment:
You can write 'foregroundProgressLayer.removeAnimation(forKey: "strokeEnd")' to remove animation. So you can remove other logic. Just leave isAnimationStarted = false